src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -419,9 +421,6 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
                 return null;
             }
 
-            // If id_token is received using code only flow, no need to validate chash.
-            await ValidateOpenIdConnectProtocolAsync(jwt, message, false);
```

Comment:
@brentschmaltz In the case, where we get a code using 'code' only flow and then redeem that for an id_token, validating that id_token will throw because the id_token does not have the chash claim.

In this line: https://github.com/aspnet/Security/blob/tratcher/oidc/src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs#L412

jwt does not have a chash claim and message.code is not null, it will throw IDX10308. The test here fails: https://github.com/aspnet/Security/blob/tratcher/oidc/src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs#L412